netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsServerHandler.java

```diff
@@ -222,7 +222,9 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
         super.handlerRemoved(ctx);
         for (ChannelHandler dependent: dependentHandlers) {
             try {
-                ctx.pipeline().remove(dependent);
+                // guard to hopefully avoid unnecessary exceptions
+                if (ctx.pipeline().context(dependent) != null)
+                    ctx.pipeline().remove(dependent);
             } catch (NoSuchElementException e) {
```

Comment:
Is the try-catch still needed after the proposed changes?

Reply:
I'd keep it just incase the context exists momentarily and then is removed in the meantime... but I'm not seeing it being caught anymore.

Reply:
Does the test suite pass w/o it?

Reply:
I've not see the test suite pass in any situation... there are some flakey tests :man_shrugging: 

I doubt the test suite would be advanced enough to catch the race condition. I've been hundreds of thousands of user requests as part of perftests recently and not seen a single exception... but who's to say it won't happen when I'm not looking?